libr/bin/p/bin_dex.c

```diff
@@ -471,44 +464,40 @@ static void dex_parse_debug_item(RBinFile *bf,
 		}
 		parameters_size--;
 	}
-
-	if (!p4 || p4 >= p4_end) {
-		free (debug_positions);
-		free (params);
-		free (debug_locals);
-		free (emitted_debug_locals);
-		free (buf);
+	ut8 opcode = 0; // = *(p4++) & 0xff;
+	if (r_buf_read (bf->buf, &opcode, 1) != 1) {
+		// error
 		return;
 	}
-	ut8 opcode = *(p4++) & 0xff;
-	while (keep && p4 + 1 < p4_end) {
+	while (keep) {//  && p4 + 1 < p4_end) {
 		switch (opcode) {
 		case 0x0: // DBG_END_SEQUENCE
 			keep = false;
 			break;
 		case 0x1: // DBG_ADVANCE_PC
 			{
 			ut64 addr_diff;
-			p4 = r_uleb128 (p4, p4_end - p4, &addr_diff);
+			// p4 = r_uleb128 (p4, p4_end - p4, &addr_diff);
```

Comment:
And this. Along with all subsequent pieces. In the event of uncertainty you can always check `git blame`.